### ISSUE TITLE
Performance Regression: PTQ-INT8 Quantization for nn.Conv2d with groups > 1 is Slower than FP16

### ISSUE THREAD

**Issue by Hyubo** [NVIDIA/TensorRT#4529]
Description
I am encountering a significant performance regression when performing Post-Training Quantization (PTQ) on a PyTorch nn.Conv2d layer where the groups parameter is greater than 1. Specifically, after converting such models to a TensorRT engine with INT8 quantization, the inference speed becomes noticeably slower than its FP16 counterpart.

Upon analyzing the generated TensorRT computation graph (using tools like trex or polygraphy's dump features), it appears that when groups > 1, TensorRT is not optimizing the grouped convolution as a single fused operation for quantization. Instead, it seems to be splitting the grouped convolution into individual, separate convolution operations for each group, and then quantizing each of these smaller, disaggregated convolutions. This disaggregation, particularly in the context of INT8 quantization, seems to introduce overhead that leads to the observed performance degradation, making the INT8 engine perform worse than the FP16 engine.

This behavior is unexpected, as INT8 quantization is generally intended to improve performance.

As an example of the issue described above, my model consists of three convolutional layers in series, with the second convolutional layer being the one exhibiting the problem (where groups > 1). I have also provided the model code and the SVG computation graph exported using trex below for visualization.

**mode code：** 
`    model = nn.Sequential(
        # the first conv layer：3->in_channels, stride=2
        nn.Conv2d(
            in_channels=in_channels,
            out_channels=in_channels,
            kernel_size=3,
            stride=2,
            padding=1,
            groups=1,
            bias=True  # 使用bias
        ),
        
        # the second conv layer：in_channels->out_channels, stride=2
        nn.Conv2d(
            in_channels=in_channels,
            out_channels=out_channels,
            kernel_size=3,
            stride=2,
            padding=1,
            groups=groups,
            bias=True  # 使用bias
        ),
        
        # the third conv layer：out_channels->128, kernel_size=1
        nn.Conv2d(
            in_channels=out_channels,
            out_channels=128,
            kernel_size=1,
            stride=1,
            padding=0,
            groups=1,
            bias=True  # 使用bias
        ),
    ).cuda()`

**SVG view：**

![Image](https://github.com/user-attachments/assets/b4f24ce8-1d99-4049-9960-f8907b447b98)

**Comment by asfiyab-nvidia** [comment]
Assign to @nvpohanh

**Comment by nvpohanh** [comment]
INT8 Convs require padding the channels-per-group to multiples of 32. Can you try aligning all channels-per-group to multiples of 32 and see if INT8 is faster than FP16 in that case?

**Comment by Hyubo** [comment]
> INT8 Convs require padding the channels-per-group to multiples of 32. Can you try aligning all channels-per-group to multiples of 32 and see if INT8 is faster than FP16 in that case?

1. I've tested the situation you described and concluded that int8 is faster than fp16 only when the convolution parameter in_channel = out_channel.
2. Furthermore, my model structure and parameters have already been trained, and I'm currently working on quantization acceleration. Some channels-per-group values are very low. For example, in one of the group convolutions in my model, the parameters in_channel = 64 and out_channel = 128. Groups = 64 ---- > channels-per-group = 2. What should I do in this situation?

**Comment by Hyubo** [comment]
> Assign to [@nvpohanh](https://github.com/nvpohanh)

May I ask if this question is still concerned, I have not received a reply for a long time

**Comment by nvpohanh** [comment]
@Hyubo Sorry, somehow I missed the previous GitHub notification email.

I think you're correct that we may have special INT8 group conv kernels when in_channel = out_channel. I will need to check internally to make sure.

> Furthermore, my model structure and parameters have already been trained, and I'm currently working on quantization acceleration. Some channels-per-group values are very low. For example, in one of the group convolutions in my model, the parameters in_channel = 64 and out_channel = 128. Groups = 64 ---- > channels-per-group = 2. What should I do in this situation?

If you cannot adjust the channel count anymore, I am afraid that you may need to stay with FP16. If you can change the channel count, please try to make sure that: (1) in_channel=out_channel (2) channels-per-group is a power of 2 (3) in_channel and out_channel are both multiples of 32 and not too smal and (4) channels-per-group is not too small. In this way you will see larger INT8 vs FP16 perf gain.

In general, we do not recommend using group conv or depthwise conv on modern GPUs because TensorCores, which is the crucial hardware to accelerate computations, have requirements on alignments. Using group/dw convs breaks the alignments so we will need to fall back to slow cuda cores or have to pad your channel sizes internally when using TensorCores, wasting some computations. Using normal convs (with channels aligning to multiples of 32) allows you to follow TensorCores alignment and get the maximum utilization of the GPU's computation power.
